internal/validators/iam/iam_validator.go

```diff
@@ -185,6 +243,51 @@ func (s *IAMRuleService) ReconcileIAMPolicyRule(rule iamRule) (*types.Validation
 	return vr, nil
 }
 
+func checkSCP(iamSvc iamApi, policyDocs []v1alpha1.PolicyDocument, policySourceArn string, policySourceType string, policySourceName string) ([]string, error) {
+	var scpFailures []string
+	var ctxEntry []iamtypes.ContextEntry
+
+	// check GetContextKeysForPrincipal to do this
```

Comment:
1. Call GetContextKeysForPrincipal to get a list of key names. The response is paginated and needs to be handled accordingly.

Reply:
2. Iterate over each context key name and resolve the value accordingly.

Implement helpers to resolve values for the following context keys:
```
All
- CurrentTime
- EpochTime

User & Role
- PrincipalArn
- PrincipalAccount
- PrincipalOrgID
- PrincipalOrgPaths

User
- Username
- Userid
```

That should be sufficient for now and we can add support for more later on.